sycl/source/detail/device_impl.cpp

```diff
@@ -284,6 +284,20 @@ bool device_impl::has(aspect Aspect) const {
                MDevice, PI_DEVICE_INFO_GPU_EU_COUNT_PER_SUBSLICE,
                sizeof(pi_device_type), &device_type,
                &return_size) == PI_SUCCESS;
+  case aspect::ext_intel_device_info_uuid: {
+    auto Result = getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+        MDevice, PI_DEVICE_INFO_UUID, sizeof(pi_device_type), &device_type,
+        &return_size);
+    if (Result != PI_SUCCESS) {
+      return false;
+    }
+
+    pi_uint8_ptr uuid =
+        static_cast<pi_uint8_ptr>(malloc(return_size * sizeof(uint8_t)));
```

Comment:
thanks for pointing this out.

Reply:
I assume that we will change the DPC++ interface back to `std::array` (see my other comment).  In that case, this code here will be a memory leak, which is bad.  If the DPC++ interface assumes that the UUID is always 16-bytes, then there is no point in trying to support an arbitrary size here.  Therefore, you could just allocate a fixed 16-byte array to receive the results from the PI query.  Something like:

```
assert(return_size <= 16);
std::byte buf[16];
/* read contents of "buf" from PI layer */
```

Reply:
Since PI interface readily supports receiving the size then  I'd still go with 2 calls. You can additionally assert here that the returned size is 16 since you'd need to truncate it to fit the array of 16-bytes returned in SYCL API.

Reply:
Sorry, could somebody please clarify the resolution here? Will return value of the SYCL device query be `std::array<uint8_t, 16>`?

Reply:
it's std::array<std::byte, 16>